content/what-is-jsx.md

```diff
@@ -0,0 +1,255 @@
+		</Fragment>
+	))}
+</dl>
+```
+
+### Trying it out
+
+You can try out how JSX translation works in the
+[Babel online compiler](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=DwWQngwg9gtgDlAdgU0QFwARoJZoDbIC8ARABLJ55QYDqUATngCbEB8AUAJDBytX3IYGbHADOAVxjAA9L3Yzw0eElRpWQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.7.3&externalPlugins=).
+It translates it using the `createElement` function from the React library,
+but we can also tell Babel to use a different `createElement` function from a
+library like complate.
+
+
+The complate JSX implementation
+-------------------------------
+
+Now we want to look at the specifics of the complate JSX implementation.
+
+We saw before that
+
+```jsx
+<MyComponent title="Hello World">
+	<p>lorem ipsum</p>
+<MyComponent>
+```
+
+becomes
+
+```javascript
+createElement(MyComponent, { title: title },
+	createElement("p", null, description));
+```
+
+In order for this to work, the complate `createElement` function needs
+to be available in the JavaScript scope.
+
+`createElement` in complate expects a user defined component (called a macro)
+to be a function with the signature `(params, ...children)` and to return a JSX
+expression. An example definition for `MyComponent` could be:
+
+```jsx
+function MyComponent({ title }, ...children) {
+    return <article>
+        <h2>{title}</h2>
+        {children}
+    </article>;
+}
+```
+
+### Dealing with boolean attributes
+
+We saw above that JSX interprets an attribute without a value as being
+the same as the boolean value true. complate also ignores the blank
+values (`false`, `undefined`, `null`) so `<button disabled={undefined}>`
+and `<button disabled={null}>` generate the same `<button>` HTML Tag
+without the `disabled` property.
+
+complate will also ensure that boolean properties are correctly set on
+HTML elements. This means that `<button disabled={true}>` will be
+correctly translated to the HTML Tag `<button disabled>`.
+
+complate also ignores blank children so that we can more easily use
+JavaScript expressions for conditionals.
+
+Here a `<p>` tag will only be generated when the description value
+is truthy.
+
+```jsx
+<article>
+	{description && <p>{description}</p>}
+<article>
+```
+
+### Multiple JSX Implementations
+
+Up until now we have skirted around the question of what the `createElement`
+function in complate actually does. This is because we decided to define the
+contract for component definitions separately from the actual implementation
+of the `createElement` functionality. Because we have optimized for simplicity
+and portability, this has allowed us to create multiple `createElement`
+implemenations which we can choose based on our use case.
+
+#### complate-stream
+
+The first complate implementation is
```

Comment:
```suggestion
The first complate implementation was
```

or 

```suggestion
The primary complate implementation was
```

I'm not actually sure either is true though... 😁 Also, I _might_ be splitting hairs here (and only here, natch).